observability/metric/meter_test.go

```
@@ -0,0 +1,216 @@
+	// Create a custom resource
+	res, err := resource.New(ctx,
+		resource.WithAttributes(),
+	)
+	require.NoError(t, err)
+
+	// Try to setup (may fail in test environment without OTLP collector)
+	mp, err := Setup(ctx, res)
+
+	if err != nil {
+		// Expected in test environment without OTLP collector
+		assert.Error(t, err)
```

golangci-lint (2.5.0)

[error] require-error: for error assertions use require

(testifylint)